### PR TITLE
VPATH fixes

### LINKED ISSUE #10071: VPATH build is busted
### Greenplum version or build
7.0 alpha / master (as of commit 36d0da913481b), but I believe this issue exists on all released versions as well.

### autoconf options used ( config.status --config )

```
# with the CWD being ~/gpdb.build.gcc10:
~/workspace/gpdb/configure --config-cache --disable-gpcloud --disable-orca --disable-gpfdist --without-readline --without-zlib --without-zstd --without-libcurl --disable-pxf --without-libbz2 --with-includes=$HOME/.opt/xerces-c-3.1/include --with-libs=$HOME/.opt/xerces-c-3.1/lib CC='ccache gcc-9' CXX='ccache g++-9' CFLAGS='-O0' CXXFLAGS='-O0' LDFLAGS=-fuse-ld=gold
```

### Expected behavior
VPATH works

### Actual behavior
All sorts of things are broken, they mostly seem to be related to misuse of `srcdir` vs `builddir`

### Step to reproduce the behavior
1. `mkdir ~/gpdb.build.gcc10`
1. `cd !$`
1. `~/workspace/gpdb/configure --config-cache --disable-gpcloud --disable-orca --disable-gpfdist --without-readline --without-zlib --without-zstd --without-libcurl --disable-pxf --without-libbz2 --with-includes=$HOME/.opt/xerces-c-3.1/include --with-libs=$HOME/.opt/xerces-c-3.1/lib CC='ccache gcc-9' CXX='ccache g++-9' CFLAGS='-O0' CXXFLAGS='-O0' LDFLAGS=-fuse-ld=gold
`
1. `make`

### PR DESCRIPTION
**Pull request by d**
### TL;DR: `srcdir`, `builddir`, potato potahto.
See either commit message for details.

### Long form: This will be a straightforward and boring review.

This fixes up quite a few confusion around the use of top_srcdir vs
top_builddir. Amusingly most of this is uncovered by running merely a
"make clean" on a VPATH build.

While we're here I've taken the liberty to fix a surrounding mistake
like CPPFLAGS vs CFLAGS, and slightly eliminate some duplication,
using Makefile inclusion.

This should let a VPATH build successfully compile (but not
link -- yet -- due to #10072).

This should resolve #10071

### CODE REVIEW

**Comment by d** [comment]
> This should let a VPATH build successfully compile (but not
> link -- yet -- due to #10072).

OK I lied, there's still one more error in `interfaces/gppc`, but I wonder whether there's any more value in fixing gppc, rather than axing it

**Comment by d** [comment]
> OK I lied, there's still one more error in `interfaces/gppc`, but I wonder whether there's any more value in fixing gppc, rather than axing it

nvm, i fixed it

**Comment by d** [comment]
> LGTM. It's easy to miss something in a big repetitive patch but having looked over it twice I can't see anything sticking out. Thanks for fixing these!

I found and fixed one more place in `gpfdist`. There are admittedly a few more places under `src/test` that I didn't test and are certainly broken but if you don't object I'm gonna leave those for a future exercise
